common/include/common/framebuffer.h

```diff
@@ -24,9 +24,23 @@
 #include <stdlib.h>
 #include <stdbool.h>
 #include <stdint.h>
+#include <stdatomic.h>
 
 typedef struct stFrameBuffer FrameBuffer;
 
+#define FB_CHUNK_SIZE   1048576 // 1MB
+#define FB_SPIN_LIMIT   10000   // 10ms
+#define FB_WP_TYPE      atomic_uint_least32_t
+#define FB_WP_SIZE      sizeof(FB_WP_TYPE)
+
+struct stFrameBuffer
```

Comment:
Might be worth merging this with the `typedef struct stFrameBuffer FrameBuffer;` above.